src/tablo/Tablo.js

```diff
@@ -0,0 +1,117 @@
+import FlexView from 'react-flexview';
+
+export const defaultWidth = 500;
+
+/** A table component based on fixed-data-table-2
+ * @param data - data shown in the table
+ * @param defaultColumns - a function that returns the default columns given the data
+ * @param width - the desired width of the table
+ * @param height - the desired height of the table
+ * @param rowHeight - height in pixel of every row
+ * @param headerHeight - height in pixel of header
+ * @param groupHeaderHeight - height in pixel of groupHeader
+ * @param footerHeight - height in pixel of footer
+ * @param onRowMouseEnter - callback to be called when mouse enters a row
+ * @param onRowMouseLeave - callback to be called when mouse leaves a row
+ * @param autosize - wheater the table should resize to fit the available space
+ * @param columnsOrder - an array containing the ordered list of column names, in the same order they should be rendered
+ * @param onColumnsReorder - callback to be called when the order of columns changes after dragging an header in a new position
+ * @param onColumnResize - callback to be called when a column is resized
+ * @param children - table children (Column or ColumnGroup)
+ * @param scrollLeft - value of horizontal scroll
+ * @param scrollTop - value of vertical scroll
+ * @param onScrollStart - callback to be called when scrolling starts
+ * @param onScrollEnd - callback to be called when scrolling ends
+ * @param selectedRows - the list of selected row ids
+ * @param onRowsSelect - callback to be called when the currently selected rows change
+ * @param selectionType - single = only one selected row at a time, multi = multiple selection, none = selection disabled
+ * @param hoveredRowIndex - the id of the hovered row
+ * @param onHoverRowChange - callback to be called when the hovered row changes
+ * @param sortBy - id of the column according which the data should be ordered
+ * @param sortDir - sorting direction
+ * @param onSortChange - callback to be called when sorting change
+ *
+ * @param scrollToRow - index of the row to scroll to
+ * @param onRowClick - callback to be called when a row is clicked
+ * @param rowClassNameGetter - called to get any additional CSS classes that should be added to a row
+ * @param onColumnResizeEndCallback - callback to be called when resizer has been released and column needs to be updated
+ * @param isColumnResizing - whether a column is currently being resized
+ */
+@skinnable()
+@pure
+@props({
+  // public
+  className: maybe(t.String),
+  data: t.Array,
+  defaultColumns: maybe(t.Function),
+  width: t.Number,
+  height: t.Number,
+  rowHeight: t.Number,
+  headerHeight: t.Number,
+  groupHeaderHeight: t.Number,
+  footerHeight: t.Number,
+  onRowMouseEnter: maybe(t.Function),
+  onRowMouseLeave: maybe(t.Function),
+  scrollLeft: maybe(t.Integer),
+  scrollTop: maybe(t.Integer),
+  onScrollStart: maybe(t.Function),
+  onScrollEnd: maybe(t.Function),
+  children: t.ReactChildren,
+
+  // private
```

Comment:
These props are used internally for plugin functionalities. Should not be overwritten by devs, or plugin functionalities will break (maybe we should filter them in plugins)